### DISCUSSION TITLE
Connect backend servers

### DISCUSSION THREAD

**Discussion by kochia3** [gravitational/teleport#17573] [Q&A] [answered]
I have an API gateway + several servers to process requests, is the teleport good option to create a overlay network between all of the backend servers without exposing them to Internet?

**Comment by webvictim** [comment, marked as answer]
If you're looking to expose HTTP/HTTPS services, Teleport application access might be a good fit.

It can also forward TCP ports in a secure way.

https://goteleport.com/docs/application-access/guides/connecting-apps/

**Reply by kochia3** [reply]
Thanks

**Reply by kochia3** [reply]
So, it's suitable for server to server connection in production?
For example to connect web-site backend to db?

**Reply by webvictim** [reply]
Machine ID can theoretically be used for that. Take a look at the docs: https://goteleport.com/docs/machine-id/introduction/
